life_expectancy/cleaning_strategies.py

```diff
@@ -0,0 +1,195 @@
+import argparse
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any
+
+import numpy as np
+import pandas as pd
+
+from life_expectancy.region import Region
+
+
+class AbstractCleaningStrategy(ABC):
+    """Abstract class that defines a strategy for cleaning some variants life_expectancy data."""
+
+    def __init__(self, col_dtypes=None):
+        super().__init__()
+        self.col_dtypes = {"year": np.int64, "value": float} if col_dtypes is None else col_dtypes
+
+    @abstractmethod
+    def load_data(self, data_path: Path, *args, **kwargs) -> pd.DataFrame:
+        """Loads data from a file in a specified path to a dataframe."""
+
+    @abstractmethod
+    def save_data(self, data: pd.DataFrame, output_path: Path | str, *args, **kwargs):
+        """Saves data to a file specified by the given path."""
+
+    @abstractmethod
+    def clean_data(self, df: pd.DataFrame, country: Region | None = None) -> pd.DataFrame:
```

Comment:
The region being  a possible "None" is because the input region couldn't be present in the pool of region inside the file?